### ISSUE TITLE
cmake outdate version of CMakeList.txt libs and third party files

### ISSUE THREAD

**Issue by Lightnet** [libsdl-org/SDL_mixer#314]
Found some minor bug in main CMakeLists.txt. It currently missing lib and dll checks. Using FetchContent build for the project.

Visual Studio 2019, window 10 64 bit, cmake 3.20.

CMakeLists.txt
```
option(BUILD_SHARED_LIBS "Build the library as a shared library" ON)
#...
if(BUILD_SHARED_LIBS)
    add_library(SDL2_mixer STATIC)
else()
    add_library(SDL2_mixer SHARED)
endif()
```
There are some outdate warning error in the log   
cmake_minimum_required(VERSION 2.8.0) > cmake_minimum_required(VERSION 3.20) that change seem to work. Well change to lower the version a bit depend if needed.

libogg-1.3.2

libvorbisidec-1.2.1

**Comment by sezero** [comment]
> There are some outdate warning error

Warning or error, which one is it?
If you post the warning (or error), we'd have a chance to see it too.

> libogg-1.3.2
> 
> libvorbisidec-1.2.1

What _exactly_ is the problem with these libraries?

**Comment by Lightnet** [comment]
It is just the cmake version checks cmake_minimum_required CMakelists.txt. One reason is that project() is empty for some reason.

**Comment by sezero** [comment]
You _still_ haven't shown us your error:  Please don't expect others to test your
projects unless you show the actual issue, e.g. some warning or error message.

That aside: The cmake support in SDL_mixer isn't even half-baked. It only targets
android and nothing else, so it will need major work to make it actually usable.

**Comment by sezero** [comment]
@icculus, @slouken: any of you interested in this? (I'm not interested in cmake stuff.)

**Comment by Lightnet** [comment]
Sorry about I busy testing the  build to get some sort working project build.

```
CMake Deprecation Warning at build/_deps/sdl_mixer-src/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning (dev) at build/_deps/sdl_mixer-src/CMakeLists.txt:2 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
    PROJECT_VERSION_PATCH
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Deprecation Warning at build/_deps/sdl_mixer-src/external/libogg-1.3.2/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning (dev) at build/_deps/sdl_mixer-src/external/libogg-1.3.2/CMakeLists.txt:2 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
    PROJECT_VERSION_PATCH
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Deprecation Warning at build/_deps/sdl_mixer-src/external/libvorbisidec-1.2.1/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning (dev) at build/_deps/sdl_mixer-src/external/libvorbisidec-1.2.1/CMakeLists.txt:2 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
    PROJECT_VERSION_PATCH
This warning is for project developers.  Use -Wno-dev to suppress it.
``` 

LINK : fatal error LNK1104: cannot open file 'Debug\SDL2_mixer.lib'

Here the error that it saying that it suppose to build SDL2_mixer.lib but instead it build SDL2_mixer.dll. Just need need to add config for develop Debug Build type.

**Comment by sezero** [comment]
Pushed 283199ed to bump cmake minimum version from 2.8.11 to 2.8.12 in some project files
(which was easy part.)

**Comment by sezero** [comment]
Looks like this has been fixed as of #336 merge.
